extension.js

```diff
@@ -171,9 +174,6 @@ class MprisLabel extends PanelMenu.Button {
 		if(SCROLL_ACTION == 'none')
 			return Clutter.EVENT_STOP
 
-		if (event.is_pointer_emulated())
-			return Clutter.EVENT_PROPAGATE;
-
```

Comment:
I'm not sure why this was initially added. It seems like my touchpad has a lot of scroll events that are considered emulated but otherwise look normal. This was causing problems for me because I have another extension with scroll event handling on the entire menu bar. A bunch of scroll events would hit this and propagate through, even though I just wanted to interact with the mpris label.
So that's why I removed it. Could also just make it return EVENT_STOP instead.

Reply:
> I'm not sure why this was initially added.

I don't remember, though [gnome-shell's code does the same](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/main/js/ui/slider.js?ref_type=heads#L156).

> This was causing problems for me because I have another extension with scroll event handling on the entire menu bar.

We found the same issue you are describing with Dash-to-panel (#47).

> So that's why I removed it. Could also just make it return EVENT_STOP instead.

I'm not sure if we want "emulated scroll events" to trigger the scrolling action, this would probably change the volume increasing/decreasing speed in some hardware (can you test if this is the case in yours?). The idea is to follow gnome's volume widget on ~behaviour, though we are changing scroll direction here...~ speed, so I'm more inclined to change EVENT_PROPAGATE to EVENT_STOP.

Reply:
I found this bugzilla thread: https://bugzilla.gnome.org/show_bug.cgi?id=687573, and these commits: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/7d4e14f38455977ffb103acd85239ed6b1a9d816, https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/c9741ae3d5f661b2aee352c50ce21fab16694fbe (the last one is somewhat confusing to me since I definitely observe events with `is_pointer_emulated()` true).
It seems that some events are essentially duplicated, one emulated and one not emulated... I don't have the historical reasoning for why this is the case but my impression is that it has to do with retrofitting touchpad support into X11.

I suppose ideally other extensions should also be checking this, but in practice it doesn't make a lot of sense to propagate the events instead of just ignoring them completely.

Upshot: happy to use EVENT_STOP, makes sense.